core/src/main/java/org/geysermc/geyser/session/GeyserSession.java

```diff
@@ -1765,11 +1765,10 @@ public void sendAdventureSettings() {
         }
 
         // https://github.com/GeyserMC/Geyser/issues/3769 Setting Spectator mode ability layer
-        if (spectator) {
-            abilityLayer.setLayerType(AbilityLayer.Type.SPECTATOR);
-        } else {
-            abilityLayer.setLayerType(AbilityLayer.Type.BASE);
-        }
+        abilityLayer.setLayerType(AbilityLayer.Type.SPECTATOR);
```

Comment:
Now we would be setting the spectator layer for all gamemodes..
The check is still needed, but there was no need for two of them